### ISSUE TITLE
Consider using staging before production against letsencrypt

### ISSUE THREAD

**Issue by Torxed** [Mailu/Mailu#3637]
It's up to the end user at the end of the day to configure things properly.
But if there's any mistake, there's a big chance you will get ratelimited against letsencrypt if there's a misconfiguration with either the firewall or DNS:

```
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='acme-v02.api.letsencrypt.org', port=443): Max retries exceeded with url: /directory (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x728fdfb5b3b0>: Failed to establish a new connection: [Errno -3] Try again'))
```

Attempting to gather certificates from their [staging environment](https://letsencrypt.org/docs/staging-environment/) first, is the recommended approach by themselves. And for good reasons.

Or I would highly suggest [swapping out `nginx` for `caddy`](https://github.com/Mailu/Mailu/issues/997) (or something else) for at least port `80` and `443` to make certificate handling slightly easier.

**Comment by nextgens** [comment]
We use [ngx_mail_core_module](http://nginx.org/en/docs/mail/ngx_mail_core_module.html); Can caddy do that?

As for using the letsencrypt staging environment by default, we welcome a PR. Get setup to add a new config variable (with a suitable comment) in `[mailu.env](https://github.com/Mailu/Mailu/blob/master/setup/flavors/compose/mailu.env)` that requires to be removed for production to be used.

Please ensure you document it in the FAQ (and if possible print out an appropriate warning messages when staging is used).

**Comment by Torxed** [comment]
No, caddy does not have protocol awareness for anything other than HTTP.
You would have to expose the mail server directly or use another reverse proxy for that. Either way, it might be worth configuring certificates another way if that's the only thing you're using nginx for.
But caddy can do tcp/udp reverse proxy.

Lets hope someone has time and knows enough about this project to submit a PR.
I only noticed a flaw/issue in this setup and thought I'd report it - I generally don't have time for another project.

**Comment by nextgens** [comment]
Very well, we'll file this where it belongs then.
